hubblestack_nova/secedit.py

```diff
@@ -0,0 +1,249 @@
+                        secedit_ret = _multi_convert(secedit_ret)
+                    if tag_data['value_type'] in ['less', 'more', 'equal']:
+                        secedit_ret = _get_operation(secedit_ret,
+                                                     tag_data['value_type'],
+                                                     tag_data['match_output'])
+
+                    found = False
+                    if secedit_ret:
+                        found = True
+                    if 'match_output' in tag_data and tag_data['match_output'] not in secedit_ret:
+                        found = False
+
+                    if found:
+                        ret['Success'].append(tag_data)
+                    else:
+                        ret['Failure'].append(tag_data)
+
+    if not verbose:
+        failure = []
+        success = []
+
+        tags_descriptions = set()
+
+        for tag_data in ret['Failure']:
+            tag = tag_data['tag']
+            description = tag_data.get('description')
+            if (tag, description) not in tags_descriptions:
+                failure.append({tag: description})
+                tags_descriptions.add((tag, description))
+
+        tags_descriptions = set()
+
+        for tag_data in ret['Success']:
+            tag = tag_data['tag']
+            description = tag_data.get('description')
+            if (tag, description) not in tags_descriptions:
+                success.append({tag: description})
+                tags_descriptions.add((tag, description))
+
+        ret['Success'] = success
+        ret['Failure'] = failure
+
+    return ret
+
+
+def _findOptionValueInSeceditFile(option):
+    '''
+    helper function to dump/parse a `secedit /export` file for a particular
+    option
+    '''
+    try:
+        _d = uuid.uuid4().hex
+        _tfile = '{0}\\{1}'.format(__salt__['config.get']('cachedir'), 'salt-secedit-dump-{0}.txt'.format(_d))
+        _ret = __salt__['cmd.run']('secedit /export /cfg {0}'.format(_tfile))
+        if _ret:
+            _reader = codecs.open(_tfile, 'r', encoding='utf-16')
+            _secdata = _reader.readlines()
+            _reader.close()
+            _ret = __salt__['file.remove'](_tfile)
+            for _line in _secdata:
+                if _line.startswith(option):
+                    return _line.split('=')[1].strip()
+        else:
+            return 'Not Defined'
+    except:
+        log.debug('error occurred while trying to get secedit data')
+        return False, None
+
+
+def _merge_yaml(ret, data):
+    '''
+    Merge two yaml dicts together at the secedit:blacklist and
+    secedit:whitelist level
+    '''
+    if 'secedit' not in ret:
+        ret['secedit'] = {}
+    for topkey in ('blacklist', 'whitelist'):
+        if topkey in data.get('secedit', {}):
+            if topkey not in ret['secedit']:
+                ret['secedit'][topkey] = {}
+            ret['secedit'][topkey].update(data['secedit'][topkey])
```

Comment:
I recommend changing this slightly. This is the general format that I used to use for the other modules, but the problem is if you have the same top-level key in two different yaml files, whichever one you parse last will overwrite the others. I switched to using a list like https://github.com/HubbleStack/Nova/blob/3de386e99734af9758a0870a0d9509c4e7df4b0e/hubblestack_nova/pkg.py#L215-L218, though that also requires some changes to _get_tags as well. You can see the way I changed all the modules in my pull request https://github.com/HubbleStack/Nova/pull/91/files

The changes look bigger than they are. In `_get_tags` you basically just need one more nested loop to iterate over the list of tag data. Keep me posted if you have questions.